src/extension.ts

```diff
@@ -261,64 +261,62 @@ export async function activate(context: vscode.ExtensionContext) {
-                [
-                    { description: "Device target (esp32, esp32s2beta)",
-                        label: "Device Target", target: "deviceTarget" },
-                    { description: "Device port path", label: "Device Port", target: "devicePort" },
-                    { description: "Baud rate of device", label: "Baud Rate", target: "baudRate" },
-                    { description: "Relative paths to OpenOCD Scripts directory separated by comma(,)",
-                        label: "OpenOcd Config Files", target: "openOcdConfig" },
-                ],
-                { placeHolder: selectConfigMsg },
-            ).then((option) => {
-                if (typeof option === "undefined") {
-                    const noOptionMsg = locDic.localize("extension.noOptionMessage",
-                        "No option selected.");
-                    Logger.infoNotify(noOptionMsg);
-                    return;
-                }
-                let currentValue;
-                let msg: string;
-                let paramName: string;
-                switch (option.target) {
-                    case "deviceTarget":
-                        msg = locDic.localize("extension.enterDeviceTargetMessage",
-                            "Enter device target name");
-                        paramName = "idf.adapterTargetName";
-                        break;
-                    case "devicePort":
-                        msg = locDic.localize("extension.enterDevicePortMessage",
-                            "Enter device port Path");
-                        paramName = "idf.port";
-                        break;
-                    case "baudRate":
-                        msg = locDic.localize("extension.enterDeviceBaudRateMessage",
-                            "Enter device baud rate");
-                        paramName = "idf.baudRate";
-                        break;
-                    case "openOcdConfig":
-                        msg = locDic.localize("extension.enterOpenOcdConfigMessage",
-                            "Enter OpenOCD Configuration File Paths list");
-                        paramName = "idf.openOcdConfigs";
-                        break;
-                    default:
-                        const noParamUpdatedMsg = locDic.localize("extension.noParamUpdatedMessage",
-                            "No device parameter has been updated");
-                        Logger.infoNotify(noParamUpdatedMsg);
-                        break;
-                }
-                if (msg && paramName) {
-                    currentValue = idfConf.readParameter(paramName);
-                    if (currentValue instanceof Array) {
-                        currentValue = currentValue.join(",");
-                    }
-                    idfConf.updateConfParameter(paramName, msg, currentValue, option.label);
+        vscode.window.showQuickPick(
+            [
+                { description: "Device target (esp32, esp32s2beta)",
```

Comment:
May be we should n't hardcode list of targets here? Recent master supports esp32s2, while esp32s2beta is only supported by 4.1 branch?

Reply:
Is there a way to get the supported targets from esp-idf ? 
Will move to a separated supported target config file.

Reply:
There is `idf.py --list-targets` command, that is supported starting v4.1

Reply:
Not sure, that this way is good enough to be used by IDE

Reply:
We still need to have some sort of target file to map it with related openOCD config files as we do here. We could have all targets in extension and show error to user if they select a non supported target for the current idf.espIdfPath, what do you think?

Reply:
I cannot come up with a better solution, so this is ok